### PR TITLE
Pr 19 CI on DB branch

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

**Comment by mattcollier** [comment]
Linting errors are preventing CI from running.  I'll fix that up tomorrow.

**Comment by mattcollier** [comment]
Hello @kdenhartog thanks again for the PR. I fixed this up to get CI running here and the tests around this new feature are failing.  Is the test suite passing for you on this branch?  It appears to me that the issuers on the RL credential and the credential are in fact the same value `did:example:123`.

See: https://github.com/digitalbazaar/vc-revocation-list/runs/971373437?check_suite_focus=true

File: tests/30-main.spec.js [hunk 1]
```diff
@@ -254,4 +255,33 @@ describe('main', () => {
...
+        id: 'urn:uuid:4886029a-7925-11ea-9274-10bf48838a41',
+        'example:test': 'foo',
+      },
+      credentialStatus: {
+        id: 'https://example.com/status/1#67342',
+        type: 'RevocationList2020Status',
+        revocationListIndex: '67342',
+        revocationListCredential: RLC.id,
+      },
+      issuer: 'did:example:123',
```
Comment (kdenhartog) [inline tests/30-main.spec.js]: This can't match the issuer did in the RLC object above. 

```suggestion
      issuer: 'did:example:1234',
```

**Review by kdenhartog — CHANGES_REQUESTED** [review]
Looks like this won't work for node 10 either because the package.json file sets the node environment to be >12.0.0. When I set [this line](https://github.com/digitalbazaar/vc-revocation-list/blob/91723fc1bbe79b79c5d18f487d49a471f0143c22/package.json#L81) to `"node": ">=10.0.0"` and made the change in the suggestion below I was able to get everything working.

**Comment by mattcollier** [comment]
@dlongley @kdenhartog would we want the default behavior to be that the verifiers should be validated?  Change the default `verifyMatchingIssuers` value to `true` and release this is a major breaking change?

**Comment by dlongley** [comment]
@mattcollier,

> would we want the default behavior to be that the verifiers should be validated? Change the default `verifyMatchingIssuers` value to `true` and release this is a major breaking change?

I'm assuming you meant "issuers" (not "verifiers"). In any case, my first inclination was to say "yes", but my second thought was that it may place an undue burden on certain decentralized use cases. An issuer may want to allow some other entity to handle revocation and publishing revocation VCs. There's nothing that's inherently less trustworthy with non-matching issuers on their own -- it's just a different mode. The question is -- do you trust your resolver to return the RC VC that the original issuer intended you to get (regardless of who signed it)?

My third thought is that we should have the defaults be the strictest they can be since this is a security lib -- and that we may end up needing to have another property in the VC itself (in the revocation section ... or define the behavior according to the credential status type) that specifies whether or not the issuer needs to match.

So, in the end, I'm +1 for requiring the issuers to match by default in this library.

**Comment by kdenhartog** [comment]
I originally took the approach of defaulting to false in order to not cause breaking changes for this, but I'm happy to see it set to true by default. @dlongley point about this being a security lib is the most compelling reason for me as well.

**Comment by mattcollier** [comment]
@dlongley @kdenhartog ready for review of breaking release.

File: main.js [hunk 2]
```diff
@@ -181,6 +193,23 @@ async function _checkStatus({
...
+    const credentialIssuer =
+      typeof credential.issuer === 'object' ?
+        credential.issuer.id : credential.issuer;
+    const revocationListCredentialIssuer =
+      typeof rlCredential.issuer === 'object' ?
+        rlCredential.issuer.id : rlCredential.issuer;
+
+    if(!(credentialIssuer && revocationListCredentialIssuer) ||
+      (credentialIssuer !== revocationListCredentialIssuer)) {
+      throw new Error('Issuers of the credentials do not match.');
```
Comment (kdenhartog) [inline main.js]:
```suggestion
      throw new Error('Issuers of the revocation credential and verifiable credential do not match.');
```

File: tests/30-main.spec.js [hunk 3]
```diff
@@ -640,4 +647,71 @@ describe('main', () => {
...
+    };
+    const result = await checkStatus({
+      credential,
+      documentLoader,
+      verifyRevocationListCredential: false,
+      verifyMatchingIssuers: true,
+    });
+    result.verified.should.equal(false);
+    result.error.message.should.equal(
+      'Issuers of the credentials do not match.');
```
Comment (kdenhartog) [inline tests/30-main.spec.js]:
```suggestion
      'Issuers of the revocation credential and verifiable credential do not match.');
```

**Review by kdenhartog — APPROVED** [review]
Added some suggested updates to the errors to make it a bit more clear what failed. Happy with either though.

**Review by dlongley — APPROVED** [review]

**Pull request merged**